.github/workflows/test-on-push.yml

```diff
@@ -0,0 +1,165 @@
+name: Super ultra fast test, lint and visual regression
+
+on: push
+
+jobs:
+  type_check:
+    name: 'Type Check and Build'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    # taken from
+    # - https://github.com/actions/cache/blob/main/examples.md#node---yarn
+    # - https://docs.github.com/en/free-pro-team@latest/actions/guides/caching-dependencies-to-speed-up-workflows
+    - name: Get yarn cache directory path
+      id: yarn-cache-dir-path
+      run: echo "::set-output name=dir::$(yarn cache dir)"
+    - uses: actions/cache@v2
+      id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+    - run: yarn --prefer-offline
```

Comment:
This is repeated for each job. We could run [matrixes](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategy) to clean this up for each script that needs to be ran.